snugget_load.py

```diff
@@ -28,25 +29,57 @@ def main():
     with conn.cursor() as cur:      
       with open(snuggetFile) as csvFile:
         newSnuggets = csv.DictReader(csvFile)
+        rowCount = 1
         for row in newSnuggets:
-          overwriteAll = processRow(appName, snuggetFile, cur, overwriteAll, row)
-  print("Snugget load complete.")
+          rowCount += 1
+          if allRequiredFieldsPresent(row, optionalFields, rowCount):
+            overwriteAll = processRow(appName, snuggetFile, cur, overwriteAll, row)
+  print("Snugget load complete. Processed", rowCount, "rows in", snuggetFile)
+
+
+
+
+
+def allRequiredFieldsPresent(row, optionalFields, rowCount):
+  if any(a != '' for a in row.values()): # if the entire row is not empty
+    blanks = []
+    for key in row.keys():
+      if (key not in optionalFields) and (row[key] == ''):
+        blanks.append(key)
+    if blanks == []:
+      return True
+    else:
+      print("Unable to process row", rowCount, "with content:")
```

Comment:
:D It already has been! Getting a bunch of new data today made it obvious how far from good enough it was to have this script just crash for issues like these.